sys/crypto/sha512.c

```diff
@@ -0,0 +1,264 @@
+/*
+ * Copyright (C) 2015 Jan Wagner <mail@jwagner.eu>
+ * Copyright (C) 2014 Daniel Beer <dlbeer@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_crypto
+ * @{
+ *
+ * @file
+ * @brief       Implementation of the SHA512 hash function
+ *
+ * @author	Jan Wagner <mail@jwagner.eu>
+ * @author	Daniel Beer <dlbeer@gmail.com>
+ * @author	NSA
```

Comment:
Question: did you use parts of a reference implementation provided by the NSA?
If yes you need to list them as authors.
Otherwise, if you are merely implementing a standard released by the NSA then you don't need
to list them as authors (AFAIK).